web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx

```diff
@@ -350,6 +371,111 @@ const CreateAccessRole = ({
   );
 };
 
+type TableData = {
+  items: SecurityGroup[];
+  nextToken?: string;
+  fetchStatus: FetchStatus;
+};
+
+const SelectSecurityGroups = ({
+  selectedSecurityGroups,
+  setSelectedSecurityGroups,
+  dbMeta,
+  emitErrorEvent,
+}: {
+  selectedSecurityGroups: string[];
+  setSelectedSecurityGroups(sg: string[]): void;
+  dbMeta: DbMeta;
+  emitErrorEvent(err: string): void;
+}) => {
+  const [sgTableData, setSgTableData] = useState<TableData>({
+    items: [],
+    nextToken: '',
+    fetchStatus: 'disabled',
+  });
+
+  const {
+    attempt: fetchSecurityGroupsAttempt,
+    setAttempt: setFetchSecurityGroupsAttempt,
+  } = useAttempt('');
+
+  function onSelectSecurityGroup(
+    sg: SecurityGroup,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) {
+    if (e.target.checked) {
+      return setSelectedSecurityGroups([...selectedSecurityGroups, sg.id]);
+    } else {
+      setSelectedSecurityGroups(
+        selectedSecurityGroups.filter(id => id !== sg.id)
+      );
+    }
+  }
+
+  async function fetchSecurityGroups() {
+    const integration = dbMeta.integration;
+    const selectedDb = dbMeta.selectedAwsRdsDb;
+
+    setFetchSecurityGroupsAttempt({ status: 'processing' });
+    try {
+      const { securityGroups, nextToken } =
+        await integrationService.fetchSecurityGroups(integration.name, {
+          vpcId: selectedDb.vpcId,
+          region: selectedDb.region,
+          nextToken: sgTableData.nextToken,
+        });
+
+      setFetchSecurityGroupsAttempt({ status: 'success' });
+      setSgTableData({
+        nextToken: nextToken,
+        fetchStatus: nextToken ? '' : 'disabled',
+        items: [...sgTableData.items, ...securityGroups],
+      });
+    } catch (err) {
+      const errMsg = getErrMessage(err);
+      setFetchSecurityGroupsAttempt({ status: 'failed', statusText: errMsg });
+      emitErrorEvent(`fetch security groups error: ${errMsg}`);
+    }
+  }
+
+  useEffect(() => {
+    fetchSecurityGroups();
+  }, []);
+
+  return (
+    <>
+      <Text bold>Step 3</Text>
+      <Text bold>Optionally Select Security Groups</Text>
+      <Text mb={2}>
+        Select security groups to assign to the Fargate service that will be
+        running the database access agent. The security groups you pick must
+        allow outbound connectivity to this Teleport cluster. If you don't
+        select any security groups, the default one for the VPC will be used.
+      </Text>
+      {fetchSecurityGroupsAttempt.status === 'failed' && (
+        <Danger>{fetchSecurityGroupsAttempt.statusText}</Danger>
+      )}
```

Comment:
we should supply a re-try button or something, otherwise if they really need to pick a security group, they'll need to refresh the page, which means to restart the flow

speaking of which, does the ec2 instance flow supply re-try buttons if it fails here too? i completely forgot 😱 . if there's any failure points where a user has to restart to get it to work, it needs a re-try button (b/c it's not a simple refresh)

if you look at RDS flows, there's varying retry options:

if upon init, we need to fetch and it fails:
<img width="827" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/87ce8a6f-a058-4966-a360-c0360cf299c9">

and a substep that fails, normally there is a "action" button, and then in turns into a "re-try" button (i saw this on ec2 instance flow)

for this case it's kinda weird that a substep has to fetch and handle two kinds of errors (one for security group and one for deployment), so instead of a big red banner, id' make it less noticeable like perhaps a textual error and a textual re-try? or anything else you think is better